### PR TITLE
Add ability to send messages in the ChatView

### PR DESCRIPTION
**Pull request by csc207jc**
Adds the ability to send a message in the chat view.

Note that the message pane is not yet updated in the ChatView.

### CODE REVIEW

File: src/interface_adapter/chat/ChatState.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package interface_adapter.chat;
+
+import interface_adapter.logged_in.LoggedInState;
+
+public class ChatState {
+    public String message = "";
+
+    public ChatState(ChatState copy) {
```
Comment (uoftm) [inline src/interface_adapter/chat/ChatState.java:8]: Should this just be message?

File: src/data_access/FirebaseMessageDataAccessObject.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public void save(Message message) {
     Request request =
         new Request.Builder()
             .url(Constants.FIREBASE_URL + "messages/" + new Date().getTime() + ".json")
-            .method("POST", RequestBody.create(stringified, MediaType.get("application/json")))
+            .method("PUT", RequestBody.create(stringified, MediaType.get("application/json")))
```
Comment (uoftm) [inline src/data_access/FirebaseMessageDataAccessObject.java:29]: Nice!

File: src/view/ChatView.java [hunk 3]
```diff
@@ -34,6 +44,35 @@ public ChatView(ChatController chatController, ChatViewModel viewModel) {
...
 
+    send.addActionListener(
+      // This creates an anonymous subclass of ActionListener and instantiates it.
+      new ActionListener() {
+        public void actionPerformed(ActionEvent evt) {
+          if (evt.getSource().equals(send)) {
+            ChatState currentState = viewModel.getState();
+
+            Message newMessage = new Message(Instant.now(), currentState.message);
+            chatController.sendMessage(newMessage);
```
Comment (uoftm) [inline src/view/ChatView.java:55]: I wonder if the UCI should handle the Instant.now construction? (this isn't necessary, but it seems like the view doesn't really need to care about time)

File: src/view/ChatView.java [hunk 4]
```diff
@@ -34,6 +44,35 @@ public ChatView(ChatController chatController, ChatViewModel viewModel) {
...
+          }
+        }
+      });
+      message.addKeyListener(
+          new KeyListener() {
+            @Override
+            public void keyTyped(KeyEvent e) {
+              ChatState currentState = viewModel.getState();
+              String text = message.getText() + e.getKeyChar();
+              currentState.message = text;
```
Comment (uoftm) [inline src/view/ChatView.java:65]: Prefer a setter function here

**Review by uoftm — COMMENTED** [review]
Reminder to format your code: `mvn com.spotify.fmt:fmt-maven-plugin:format`

File: src/interface_adapter/chat/ChatState.java (hunk 1 above)
Comment (csc207jc) [reply src/interface_adapter/chat/ChatState.java]: It's possible that we may add more fields to ChatState in the future.
In addition, this is the paradigm established by our existing code.

Comment (uoftm) [reply src/interface_adapter/chat/ChatState.java]: You're right... But it isn't used, so why can't we just do .clone?

File: src/view/ChatView.java (hunk 4 above)
Comment (csc207jc) [reply src/view/ChatView.java]: Fixed by adding getter/setter to ChatState

File: src/view/ChatView.java (hunk 3 above)
Comment (csc207jc) [reply src/view/ChatView.java]: Fixed by refactoring `Instant.now()` into the use case interactor

**Review by uoftm — APPROVED** [review]
Looks good!

**Pull request merged**